src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,649 @@
+                    + amazonAccountDetail.getAwsAccount());
+            } else {
+                valid = processExternalBill(amazonAccountDetail) && valid;
+            }
+            count ++;
+        }
+        LOG.info("Processed " + count + " AWS records");
+        return valid;
+    }
+
+    private boolean processExternalBill(AmazonAccountDetail amazonAccountDetail) {
+        boolean valid = true;
+        try {
+            TransactionDTO transactionDTO = buildTransactionDTO(amazonAccountDetail);
+            if (transactionDTO.isFailingError()) {
+                valid = false;
+                LOG.error("processAmazonAccountDetails() there errors that prevent a creation of a DI " + amazonAccountDetail.toString());
+            } else if (shouldBuildDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO.getAmount())) {
+                buidAndRouteDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO);
+            }
+        } catch (Throwable e) {
+            valid = false;
+            LOG.error("processAmazonAccountDetails() there was a problem building a DI for " + amazonAccountDetail.toString(), e);
+            GlobalVariables.getMessageMap().getErrorMessages().clear();
+        }
+        return valid;
+    }
+    
+    protected TransactionDTO buildTransactionDTO(AmazonAccountDetail amazonAccountDetail) {
+        TransactionDTO transactionDTO = new TransactionDTO();
+        transactionDTO.setLineDescription(buildAccountingLineDescription());
+        
+        transactionDTO.setAmount(convertCostStringToKualiDecimal(amazonAccountDetail.getCost()));
+        if (transactionDTO.getAmount().isLessEqual(KualiDecimal.ZERO)) {
+            transactionDTO.setFailingError(true);
+        }
+        
+        if (StringUtils.contains(amazonAccountDetail.getCostCenter(), KFSConstants.DASH)) {
+            //could be account & subaccount or whole transaction string
+            String[] costCenterArray = StringUtils.split(amazonAccountDetail.getCostCenter(), KFSConstants.DASH);
+            if (costCenterArray.length == 2) {
+                updateTransactionDTOAccountSubAccount(amazonAccountDetail, transactionDTO);
+            } else if (costCenterArray.length == 7) {
+                updateTransactionDTOFullAccountString(amazonAccountDetail, transactionDTO);
+            } else {
+                transactionDTO.setFailingError(true);
+                LOG.error("buildAmazonDistributionOfIncomeTransactionDTO: There was an unexpected number of alements in the cost center attribute, unable to processs");
+            }
+        } else {
+            updateTransactionDTOAccountNumber(amazonAccountDetail, transactionDTO);
+        }
+        
+        return transactionDTO;
+    }
+    
+    protected KualiDecimal convertCostStringToKualiDecimal(String costString) {
+        KualiDecimal cost = KualiDecimal.ZERO;
```

Comment:
There's no code that already does this elsewhere in the system? Seems pretty generic.